File: app/src/main/java/ru/strcss/projects/moneycalc/moneycalcandroid/componets/home/HomeContract.java
package ru.strcss.projects.moneycalc.moneycalcandroid.componets.home;

import java.util.List;

import ru.strcss.projects.moneycalc.moneycalcandroid.BasePresenter;
import ru.strcss.projects.moneycalc.moneycalcandroid.BaseView;
import ru.strcss.projects.moneycalc.moneycalcdto.dto.crudcontainers.statistics.FinanceSummaryFilter;
import ru.strcss.projects.moneycalc.moneycalcdto.entities.FinanceSummaryBySection;

/**
 * This specifies the contract between the view and the homePresenter.
 */
public interface HomeContract {
    interface View extends BaseView<Presenter> {

        void showDatesRange(String from, String to);

        void showStatisticsSections(List<FinanceSummaryBySection> financeSummary);

        void showAddTransactionActivity();

        void updateStatsAndSettings();
    }

    interface Presenter extends BasePresenter<View> {
        void requestFinanceSummary(FinanceSummaryFilter financeSummaryGetContainer);

        void requestSettings();

        void requestSpendingSections();

        void requestSectionStatistics();

        void requestSectionStatistics(String from, String to, List<Integer> sections);

        void showAddTransactionActivity();

        //Stats
//        void saveStatsFragmentState(int sectionId, android.view.View view);
//
//        void clearStatsFragmentStates();
//
//        void updateStatisticsSection(Integer id);
    }
}

File: app/src/main/java/ru/strcss/projects/moneycalc/moneycalcandroid/componets/home/HomeContract.kt
package ru.strcss.projects.moneycalc.moneycalcandroid.componets.home

import ru.strcss.projects.moneycalc.moneycalcandroid.BasePresenter
import ru.strcss.projects.moneycalc.moneycalcandroid.BaseView
import ru.strcss.projects.moneycalc.moneycalcdto.entities.statistics.SummaryBySection

/**
 * This specifies the contract between the view and the homePresenter.
 */
interface HomeContract {
    interface View : BaseView<Presenter> {

        fun showDatesRange(from: String, to: String)

        fun showStatisticsSections(financeSummary: List<SummaryBySection>?)

        fun showAddTransactionActivity()

        fun updateStatsAndSettings()

        fun redrawTabLogos()
    }

    interface Presenter : BasePresenter<View> {
        fun requestSettings()

        fun requestSpendingSections()

        fun requestStatsBySectionSummary()

//        fun requestSectionStatistics(from: String, to: String, sections: List<Int>)

        fun showAddTransactionActivity()
    }
}
